packages/data/src/components/with-select/index.js

```diff
@@ -9,6 +9,44 @@ import { createHigherOrderComponent } from '@wordpress/compose';
  * Internal dependencies
  */
 import { RegistryConsumer } from '../registry-provider';
+import { AsyncModeConsumer } from '../async-mode-provider';
+
+const waitingList = [];
+const componentsMap = new WeakMap();
+let isRunning = false;
+const runWaitingList = () => {
+	if ( waitingList.length === 0 ) {
+		isRunning = false;
+		return;
+	}
+	const nextComponent = waitingList.shift();
+	componentsMap.get( nextComponent )();
+	componentsMap.delete( nextComponent );
+	window.requestAnimationFrame( runWaitingList );
+};
+
+const addToWaitingList = ( component, item ) => {
+	if ( ! componentsMap.has( component ) ) {
+		waitingList.push( component );
+	}
+	componentsMap.set( component, item );
+	if ( ! isRunning ) {
+		isRunning = true;
+		window.requestAnimationFrame( runWaitingList );
```

Comment:
Same comment as above; I don't think `requestAnimationFrame` is likely to be what you want here.